ragna/deploy/_ui/modal_configuration.py

```diff
@@ -296,42 +334,71 @@ def toggle_card(event):
             else:
                 toggle_button.name = toggle_button.name.replace("▶", "▼")
 
-        toggle_button = pn.widgets.Button(
-            name="Advanced Configurations   ▶",
-            button_type="light",
-            css_classes=["modal_configuration_toggle_button"],
-        )
-
         toggle_button.on_click(toggle_card)
 
-        toggle_button.js_on_click(
-            args={"card": card},
-            code=js.JS_TOGGLE_CARD,
-        )
-
         return pn.Column(toggle_button, card)
 
+    @pn.depends(
+        "corpus_or_upload",
+        "config",
+        "config.assistant_name",
+        "config.source_storage_name",
+    )
+    def corpus_or_upload_config(self):
+        if self.corpus_or_upload == USE_CORPUS_LABEL:
+            return self.advanced_config(is_corpus=True)
+        else:
+            return self.advanced_config(is_corpus=False)
+
+    @pn.depends("advanced_config_collapsed", watch=True)
```

Comment:
Watches for when the advanced config is expanded, and then reduces the size of the file upload container (see comments in `file_uploader.py`)